core/src/workflows/author_update_entry.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    agent::actions::commit::commit_entry,
+    context::Context,
+    network::actions::publish::publish,
+    nucleus::{
+        actions::build_validation_package::build_validation_package, validation::validate_entry,
+    },
+};
+
+use holochain_core_types::{
+    cas::content::{Address, AddressableContent},
+    entry::Entry,
+    error::HolochainError,
+    validation::{EntryAction, EntryLifecycle, ValidationData},
+};
+use std::sync::Arc;
+
+pub async fn author_update_entry<'a>(
+    entry: &'a Entry,
+    maybe_link_update_delete: Option<Address>,
```

Comment:
I think for this function, the `link_update_delete` shouldn't be an Option.  I don't think you should be able to call author_update_entry without actually specifying which other entry is being updated.  It does mean you'll have to add the `Some()` around the link in the call to `commit` in step 3

Reply:
Yes, I agree. I am also wondering why we need a separate workflow. Can we not reuse `author_entry`?

Looking at this I notice that the only thing different is `EntryAction::Modify` in the validation data, which is correct. But since the `remove_entry` action also calls `author_entry` the action in the validation data will be wrong in the deletion case..